azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -794,6 +800,13 @@ func resourceArmKubernetesClusterCreate(d *schema.ResourceData, meta interface{}
 		}
 	}
 
+	if v, ok := d.GetOk("private_dns_zone_id"); ok {
+		if parameters.Identity.Type == containerservice.ResourceIdentityTypeSystemAssigned {
```

Comment:
an `identity` block is _technically_ optional at this point (although it's required in future) where you can still have a cluster using a Service Principal - so is it worth inverting this:

```suggestion
		if parameters.Identity == nil || parameters.Identity.Type != containerservice.ResourceIdentityTypeUserAssigned {
```